### PR TITLE
feat(react-native-cli): Added --verbose option (#1184)

### LINKED ISSUE #1184: Add `--verbose` option to `react-native run-windows` command
We often get issues filed for cases where NuGet fails, or MSBuild fails, and currently run-windows doesn't give much insight into the root cause of the failure.  We should add a --verbose flag to allow people to file issues with more details.

### PR DESCRIPTION
**Pull request by sayar**
Added the --verbose option which will now pipe the output of all spawned or executed commands.

Fixes #1184

### CODE REVIEW

File: local-cli/runWindows/utils/build.js [hunk 1]
```diff
@@ -18,13 +18,14 @@ function buildSolution(slnFile, buildType, buildArch) {
...
+  msBuildTools.buildProject(slnFile, buildType, buildArch, null, verbose);
 }
 
-function restoreNuGetPackages(options, slnFile) {
+function restoreNuGetPackages(options, slnFile, verbose) {
   console.log(chalk.green('Restoring NuGet packages'));
   const nugetPath = options.nugetPath || path.join(options.root, 'node_modules/react-native-windows/local-cli/runWindows/.nuget/nuget.exe');
-  const results = execSync(`"${nugetPath}" restore "${slnFile}" -NonInteractive`).toString().split(EOL);
+  const execOptions = verbose ? { stdio: 'inherit' }: {};
+  const results = execSync(`"${nugetPath}" restore "${slnFile}" -NonInteractive`, execOptions).toString().split(EOL);
```
Comment (rozele) [inline local-cli/runWindows/utils/build.js]: >   [](start = 0, length = 2)

Do you think we should use the `-Verbosity` flag here?

**Comment by rozele** [comment]
>       '/nologo',

Should we change /clp:Verbosity -> diag?

---
Refers to: local-cli/runWindows/utils/msbuildtools.js:32 in 1d0a43b. [](commit_id = 1d0a43bea7dc9906c77aef8d3d7d3ac00335a935, deletion_comment = False)

**Comment by rozele** [comment]
>       '/nologo',

Or at the very least, not minimal

---
In reply to: [317598092](https://github.com/Microsoft/react-native-windows/pull/1226#issuecomment-317598092) [](ancestors = 317598092)

---
Refers to: local-cli/runWindows/utils/msbuildtools.js:32 in 1d0a43b. [](commit_id = 1d0a43bea7dc9906c77aef8d3d7d3ac00335a935, deletion_comment = False)

File: local-cli/runWpf/utils/build.js [hunk 2]
```diff
@@ -18,13 +18,14 @@ function buildSolution(slnFile, buildType, buildArch) {
...
-  msBuildTools.buildProject(slnFile, buildType, buildArch, null);
+  msBuildTools.buildProject(slnFile, buildType, buildArch, null, verbose);
 }
 
 function restoreNuGetPackages(options, slnFile) {
   console.log(chalk.green('Restoring NuGet packages'));
   const nugetPath = options.nugetPath || path.join(options.root, 'node_modules/react-native-windows/local-cli/runWpf/.nuget/nuget.exe');
-  const results = execSync(`"${nugetPath}" restore "${slnFile}" -NonInteractive`).toString().split(EOL);
+  const execOptions = options.verbose ? { stdio: 'inherit' }: {};
+  const results = execSync(`"${nugetPath}" restore "${slnFile}" -NonInteractive`, execOptions).toString().split(EOL);
```
Comment (rozele) [inline local-cli/runWpf/utils/build.js]: >execSync [](start = 18, length = 8)

If we change for UWP, we should also set -Verbosity here.

**Comment by rozele** [comment]
>       '/clp:NoSummary;NoItemAndPropertyList;Verbosity=minimal',

If we change for UWP, we should also set for WPF. /clp:Verbosity > minimal

---
Refers to: local-cli/runWpf/utils/msbuildtools.js:31 in 1d0a43b. [](commit_id = 1d0a43bea7dc9906c77aef8d3d7d3ac00335a935, deletion_comment = False)

File: local-cli/runWindows/utils/build.js (hunk 1 above)
Comment (sayar) [inline local-cli/runWindows/utils/build.js]: Sure, I'll set it to "-Verbosity normal" as "detailed" is too much.

**Comment by sayar** [comment]
@rozele I made all the changes you suggested above and made those commands output to stdio by default although with the verbosity controlled through the --verbose flag.

**Comment by sayar** [comment]
Updated this to latest master branch (as of 7/27 3pm).

**Pull request merged**